specs/core/0_beacon-chain.md

```diff
@@ -805,17 +816,24 @@ def get_randao_mix(state: BeaconState,
     return state.latest_randao_mixes[epoch % LATEST_RANDAO_MIXES_LENGTH]
 ```
 
-### `get_active_index_root`
+### `get_active_registry_root`
 
 ```python
-def get_active_index_root(state: BeaconState,
-                          epoch: Epoch) -> Bytes32:
+def get_active_registry_root(state: BeaconState) -> Bytes32:
     """
-    Return the index root at a recent ``epoch``.
-    ``epoch`` expected to be between
-    (current_epoch - LATEST_ACTIVE_INDEX_ROOTS_LENGTH + ACTIVATION_EXIT_DELAY, current_epoch + ACTIVATION_EXIT_DELAY].
+    Return the active registry root for the current epoch.
     """
-    return state.latest_active_index_roots[epoch % LATEST_ACTIVE_INDEX_ROOTS_LENGTH]
+    epoch = get_current_epoch(state)
+    active_registry = ActiveRegistry()
+    for committee_number in range(get_epoch_committee_count(state, epoch)):
+        shard = (get_start_shard(state, epoch) + committee_number) % SHARD_COUNT
+        for index in get_crosslink_committee(state, epoch, shard):
+            validator = validator_registry[index]
+            active_registry[shard].pubkeys.append(validator.pubkey)
+            # index (top 7 bytes) + slashed (8th bit) + effective_balance (bottom 7 bits)
+            compact_validator = index << 8 + validator.slashed << 7 + validator.effective_balance // GWEI_PER_ETH
```

Comment:
This does seem like a departure from our usual style. So far we have avoided the efficiencies of shaving off bytes even in places where they look very significant (balances modulo 1 ETH) to preserve our "purity" of just using nice clean uint64s for everything. Here, doing the same thing would only increase the byte count from 8 to 17 (8 for index, 8 for balance, 1 for slashed). What's the rationale for doing it differently here?

Reply:
> This does seem like a departure from our usual style.

Right. Below are my reasons (I don't feel too strongly):

1) It's a hashing saving (no individual container leaves, and four `uint64` get packed per SSZ leaf, as opposed to one `container`). This may seem like a small saving for the beacon chain, but it could be significant for ultra-light clients (e.g. Eth2.0 client in Eth1.0).
2) Light clients are "extra-consensus", in the sense that `latest_active_registry_roots ` is exposed but never read by the state transition function.